packages/bootstrap-4/src/DescriptionField/DescriptionField.tsx

```diff
@@ -1,24 +1,22 @@
-import React from 'react';
+import React from "react";
 
-import { FieldProps } from '@rjsf/core';
+interface DescriptionFieldI  {
+    description? : string
```

Comment:
Do we know why the `description` optional? In what scenario do we want to support rendering `<DescriptionField />` without a description? 

Is this just to mirror the material-ui behaviour? To me it would make more sense to make the prop optional and fix this in material-ui instead. 

@epicfaace, thoughts?

Reply:
The bahavior is [similar to `material-ui`](https://github.com/MLH-Fellowship/react-jsonschema-form/blob/master/packages/material-ui/src/DescriptionField/DescriptionField.tsx)

Though I dont know why its optional !

Reply:
Actually, I don't think the material-ui theme defines its own version of `DescriptionFieldProps` -- it just uses the generic `FieldProps` type from `@rjsf/core`.

see https://github.com/MLH-Fellowship/react-jsonschema-form/blob/master/packages/material-ui/src/DescriptionField/DescriptionField.tsx and https://github.com/rjsf-team/react-jsonschema-form/blob/b4824b9f44b84fced7187bc752d8614a3c90b72c/packages/core/index.d.ts#L120

It might be good just to be consistent with that and not define extra types in this theme, though we could certainly add `description?: string` to the `FieldProps` type in `index.d.ts`.

Reply:
I created the new interface cause it was only using the `description` prop. Though it makes sense to use the generic and change the `index.d.ts` .

Reply:
@epicfaace I'm not sure if I agree with using `FieldProps` here, maybe you could clarify the intent? Is the idea that every RJSF component should take in an uniform prop interface, whether or not they support or use them?

As I see it, the FieldProps are not applicable, as none of them are actually passed to the h2 element, and even if they were, they wouldn't be valid HTML. So if someone passes [any of the wide variety of props on FieldProps](https://github.com/rjsf-team/react-jsonschema-form/blob/b4824b9f44b84fced7187bc752d8614a3c90b72c/packages/core/index.d.ts#L120-L142) to this element, they will simply get ignored.

Is there a reason to not define more specific prop types for elements when we know what the component actually accepts?

Reply:
@jevakallio I agree with you regarding defining the prop types for description within DescriptionField itself, but maybe a case (maybe a brand new widget) where the description type is used again, then there would be a repetition of code. In this case, I think having a common ground in FieldProps would help with implementing newer widgets using the description prop.

Reply:
Personally, I wouldn't worry too much about repetition of code. Generally, the DRY principle is often used way too eagerly. I'd rather have a couple of different places where we duplicate a `description?: string` field, than to try to optimise too soon and create a reusable `DescriptionProps` type with just a single property that's shared in a bunch of places, and then if some of the components suddenly need another prop, to have to de-duplicate them again.

However I think we should follow what @epicfaace thinks, since he has best context on the project's requirements.

Reply:
>  In this case, I think having a common ground in FieldProps would help with implementing newer widgets using the description prop.

We can do this, but we should be aware that this is semantically unclear. What does `description` prop mean in the context of a Button widget? Or any other widget? And is it confusing if a user sees it in their TS autocompletions, passes it in, and it doesn't get applied anywhere?

Reply:
> Is there a reason to not define more specific prop types for elements when we know what the component actually accepts?

Not really. We could definitely go with this; I was just pointing out that `DescriptionFieldProps` isn't present in the other themes, so it would also be fine just to be consistent with the other themes and use `FieldProps`. We could also create our own interface `DescriptionFieldProps` that perhaps extends `FieldProps`, but just using `FieldProps` would be the bare minimum.

> Personally, I wouldn't worry too much about repetition of code. Generally, the DRY principle is often used way too eagerly. I'd rather have a couple of different places where we duplicate a `description?: string` field, than to try to optimise too soon and create a reusable `DescriptionProps` type with just a single property that's shared in a bunch of places, and then if some of the components suddenly need another prop, to have to de-duplicate them again.

If we were to create an `DescriptionFieldProps` interface, we would probably want all themes to use that interface as well, and export it from the `@rjsf/core` types -- so doing this is probably a bigger refactoring task that is bigger than the scope of just adding the bootstrap 4 theme. For now, in the context of the current theme, I don't mind whether you create your own named interface or just define an inline type for this.

Reply:
Thanks Ashwin! 

@Xtremilicious sounds like it's up to you to decide whether to use the generic fieldprops or to create (or use inlined) `DescriptionFieldProps` for this. Let's pick one approach for the whole theme and go with that.

Reply:
@anikethsaha What do you think we should go with, I think creating our own interface `DescriptionFieldProps` which extends `FieldProps` seems to be the right mixture?

Reply:
I guess we can go with the extend this as you mentioned.